src/Grpc.AspNetCore.Server/Internal/CallHandlers/ServerCallHandlerBase.cs

```diff
@@ -55,7 +56,14 @@ public Task HandleCallAsync(HttpContext httpContext)
         {
             if (GrpcProtocolHelpers.IsInvalidContentType(httpContext, out var error))
             {
-                GrpcProtocolHelpers.SendHttpError(httpContext.Response, StatusCodes.Status415UnsupportedMediaType, StatusCode.Internal, error!);
+                GrpcProtocolHelpers.SendHttpError(httpContext.Response, StatusCodes.Status415UnsupportedMediaType, StatusCode.Internal, error);
+                return Task.CompletedTask;
+            }
+            if (httpContext.Request.Protocol != GrpcProtocolConstants.Http2Protocol)
+            {
+                var protocolError = $"Request protocol '{httpContext.Request.Protocol}' is not supported.";
+                GrpcProtocolHelpers.SendHttpError(httpContext.Response, StatusCodes.Status426UpgradeRequired, StatusCode.Internal, protocolError);
+                httpContext.Response.Headers[HeaderNames.Upgrade] = GrpcProtocolConstants.Http2Protocol;
```

Comment:
@Tratcher @halter73 Is 426 the right status to return when a HTTP/1.1 request comes to a gRPC?

Reply:
Anything you put down will be primarily for diagnostic purposes, there's no automatic transition support beyond ALPN.
426 is OK, 421 looks like the second best option.

Reply:
Diagnostics and providing a clear error message is the goal here. Currently HTTP/1.1 requests get through to the gRPC request delegate and only fail when the framework tries to get or set trailers.

Will use 426.